File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    let ret = true;
    const stk = [];
    
    for (let i = 0; i < s.length; i++){
        if(stk.length > 0 && stk[stk.length - 1] == '(' && s[i] == ')') stk.pop();
        else stk.push(s[i]);
    }
    
    ret = stk.length === 0 ? true : false;

    return ret;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include <bits/stdc++.h>
using namespace std;

bool solution(string s)
{
    bool answer = true;

   stack<char> stk;
    
    for (char c : s){
        if(!stk.empty() && stk.top() == '(' && c == ')') stk.pop();
        else stk.push(c);
    }
    
    answer = stk.empty() ? true : false;

    return answer;
}